packages/elements/src/internals/machines/utils/formatters.ts

```diff
@@ -0,0 +1,24 @@
+import { titleize } from '@clerk/shared';
+
+export function formatName(...args: (string | undefined)[]): string | undefined {
+  switch (args.length) {
+    case 0:
+      return undefined;
+    case 1:
+      return titleize(args[0]);
+    default:
+      return args.filter(Boolean).map(titleize).join(' ');
+  }
+}
+
+export function formatSalutation({
+  firstName,
+  lastName,
+  identifier,
+}: {
+  firstName: string | undefined;
+  lastName: string | undefined;
+  identifier: string | undefined;
+}): string {
+  return (firstName && formatName(firstName)) || (lastName && formatName(lastName)) || identifier || '';
+}
```

Comment:
💭 this is potentially something that we'd want to get from FAPI instead.

Reply:
Definitely, though in the mean time... 🤷

Reply:
Maybe we can leave a comment as a breadcrumb? 😄 e.g.

```
// TODO: ideally the derivation of these values lives in FAPI and comes back directly from the API
```